src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -10934,15 +10994,6 @@ private static bool IsNullabilityMismatch(TypeSymbol type1, TypeSymbol type2)
         public override BoundNode? VisitQueryClause(BoundQueryClause node)
         {
             var result = base.VisitQueryClause(node);
-
-            if (node.Value is BoundLambda lambda)
```

Comment:
Are there other places like this one that can now be simplified?

Reply:
The call in `VisitArguments` (line 6273) is the one I'm wondering about (you'd mentioned that place in previous PR). I didn't quite follow. 
Why can't we remove that special case (the whole `if (argument is not BoundConversion && argumentNoConversion is BoundLambda lambda)` block)?

Reply:
> Why can't we remove that special case (the whole `if (argument is not BoundConversion && argumentNoConversion is BoundLambda lambda)` block)?

We could do that, but when the argument is processed by `public VisitLambda`, we don't know if it will be also processed later by `VisitConversion` and hence `private VisitLambda`. That depends not only on the `if (argument is not BoundConversion && argumentNoConversion is BoundLambda lambda)` but also on the other condition below (`parameter is not null`) and on behavior of `VisitArgumentConversionAndInboundAssignmentsAndPreConditions` (which seems to be that the conversion is visited when ref kind is `in` or none). I could probably replicate all this logic where the `public VisitLambda` will be called (inside `VisitArgumentsEvaluate` on line 6201), but that seems fragile?

Reply:
Thanks. I agree that replicating that logic is undesirable